src/vast-ima-player.ts

```diff
@@ -226,41 +243,19 @@ export class Player extends DelegatedEventTarget {
       options.disableCustomPlaybackForIOS10Plus
     );
 
-    this.#adDisplayContainer = new ima.AdDisplayContainer(
-      adElement,
-      // used as single element for linear ad playback on iOS
-      options.disableCustomPlaybackForIOS10Plus ? undefined : mediaElement,
-      // allows to override the 'Learn More' button on mobile
-      options.clickTrackingElement
-    );
-    this.#adElement.style.display = 'none';
-    this.#adsLoader = new ima.AdsLoader(this.#adDisplayContainer);
-
     // later used for to determine playhead for triggering midrolls
     // and to determine whether the player is currently playing content
     this.#customPlayhead = new CustomPlayhead(this.#mediaElement);
     this._handleMediaElementEvents = this._handleMediaElementEvents.bind(this);
     MEDIA_ELEMENT_EVENTS.forEach((eventName) => {
       this.#mediaElement.addEventListener(
-        eventName, this._handleMediaElementEvents
+        eventName,
+        this._handleMediaElementEvents
       );
     });
 
     this._onAdsManagerLoaded = this._onAdsManagerLoaded.bind(this);
 
-    this.#adsLoader.addEventListener(
-      this.#ima.AdsManagerLoadedEvent.Type.ADS_MANAGER_LOADED,
-      this._onAdsManagerLoaded,
-      false
-    );
-    this.#adsLoader.addEventListener(
-      this.#ima.AdErrorEvent.Type.AD_ERROR,
-      (event) => {
-        this._onAdError(this._createPlayerErrorFromImaErrorEvent(event))
-      },
-      false
-    );
-
```

Comment:
This whole block together with this https://github.com/glomex/vast-ima-player/pull/41/files#diff-fde5d89523991b47e07c73406e95e095dc09dac22961938fcdf65634a96592c8L229-L238 got moved into `_resetIma` method (https://github.com/glomex/vast-ima-player/pull/41/files#diff-fde5d89523991b47e07c73406e95e095dc09dac22961938fcdf65634a96592c8R590). `reset` and then `_resetIma` always gets called when `playAds` or `loadAds` gets called.